src/doc/style-guide/src/expressions.md

```diff
@@ -663,7 +663,8 @@ never use a block (unless the block is empty).
 
 If the right-hand side consists of multiple statements, or has line comments,
 or the start of the line does not fit on the same line as the left-hand side,
-use a block.
+use a block. Do not flatten a right-hand side block containing a single macro call
+because its expanded form could contain a trailing semicolon.
```

Comment:
I'm also not 100% convinced the "why" clause of this sentence is necessary and don't feel too strongly about it. I just thought it would be a helpful addition given the context, and there are a handful of other places in the guide where a brief explanation is included with the prescription

Reply:
I think the "why" is important, since the reason for this rule is not at all obvious.